packages/client/src/reducers/contacts.js

```diff
@@ -0,0 +1,44 @@
+/*
+ * contacts is an object of { <treasury address>: [contact objects] }
+ * where each contact object has { name: <string>, address: <string> }
+ */
+export const INITIAL_STATE = {
+  contacts: {},
+};
+
+const reducer = (state, action) => {
+  const address = Object.keys(action.payload)[0];
+  const body = action.payload[address];
+  const treasuryContacts = (state.contacts?.[address] ?? []).slice(0);

```

Comment:
We want to copy the state array instead of referring to it to maintain the immutibility
```suggestion
  const treasuryContacts = ([...state.contacts?.[address]] ?? []).slice(0);

```

Reply:
that's why I have `.slice(0)` which makes a copy of the array - kind of an old school method before rest spread became the norm